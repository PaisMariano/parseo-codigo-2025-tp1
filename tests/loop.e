class LOOP_EXAMPLE
feature
    count_down
        local
            i: INTEGER
        do
            from
                i := 5
            until
                i = 0
            loop
                print (i.out)
                i := i - 1
            end
        end
end
