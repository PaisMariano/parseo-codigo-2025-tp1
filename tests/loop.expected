TOKEN_CLASS -> class
TOKEN_IDENTIFIER -> LOOP_EXAMPLE
TOKEN_FEATURE -> feature
TOKEN_IDENTIFIER -> count_down
TOKEN_LOCAL -> local
TOKEN_IDENTIFIER -> i
TOKEN_COLON -> :
TOKEN_IDENTIFIER -> INTEGER
TOKEN_DO -> do
TOKEN_FROM -> from
TOKEN_IDENTIFIER -> i
TOKEN_ASSIGN -> :=
TOKEN_NUMBER -> 5
TOKEN_UNTIL -> until
TOKEN_IDENTIFIER -> i
TOKEN_EQ -> =
TOKEN_NUMBER -> 0
TOKEN_LOOP -> loop
TOKEN_IDENTIFIER -> print
TOKEN_LPAREN -> (
TOKEN_IDENTIFIER -> i
TOKEN_DOT -> .
TOKEN_IDENTIFIER -> out
TOKEN_RPAREN -> )
TOKEN_IDENTIFIER -> i
TOKEN_ASSIGN -> :=
TOKEN_IDENTIFIER -> i
TOKEN_MINUS -> -
TOKEN_NUMBER -> 1
TOKEN_END -> end
TOKEN_END -> end
TOKEN_END -> end
